brainvisa/toolboxes/spm/processes/tools/SPMtissuesvolume.py

```diff
@@ -120,183 +127,105 @@ def initialization(self):
-            tmpfile = context.temporary('gz compressed NIFTI-1 image')
-            aims.write(tiv_mask, tmpfile)
-            context.system( 'AimsConnectComp',
-                            '-i', tmpfile,
-                            '-o', tmpfile,
-                            '-c', 26, '-n', 1)
-            context.system( 'AimsMorphoMath',
-                            '-i', tmpfile,
-                            '-o', tmpfile,
-                            '-m', 'clo', '-r', 1)
-            mask = aims.read(tmpfile)
-            mask_arr = mask.arraydata()
-            
-            wm_arr[mask_arr == 0] = False
-            gm_arr[mask_arr == 0] = False
-            csf_arr[mask_arr == 0] = False
-            tiv_arr[mask_arr == 0] = False
-            #Calculation of the volumes
-            wm_vol = wm_arr.sum()*vox_vol/1000.
-            gm_vol = gm_arr.sum()*vox_vol/1000.
-            csf_vol = csf_arr.sum()*vox_vol/1000.
-            tiv = tiv_arr.sum()*vox_vol/1000.
+        tiv = compute_vol(tiv)
+        wm_vol = compute_vol(white)
+        gm_vol = compute_vol(grey)
+        csf_vol = compute_vol(csf)
 
-        #Write in the output file
-        f.write(subject+","+
-                str(round(gm_vol,3))+","+
-                str(round(wm_vol,3))+","+
-                str(round(csf_vol,3))+","+
-                str(round(tiv,3))+"\n")
+    elif self.mode == 'proba_max':
+        tmp_w = context.temporary('gz compressed NIFTI-1 image')
+        context.runProcess('computeOneTissueMask',
+                           native_prob_map=self.white_proba_map,
+                           others_prob_maps=[self.grey_proba_map, self.csf_proba_map,
+                                             self.skull_proba_map, self.scalp_proba_map],
+                           native_mask=tmp_w,
+                           resolve_equal_probability=True)
+        tmp_g = context.temporary('gz compressed NIFTI-1 image')
+        context.runProcess('computeOneTissueMask',
+                           native_prob_map=self.grey_proba_map,
+                           others_prob_maps=[self.white_proba_map, self.csf_proba_map,
+                                             self.skull_proba_map, self.scalp_proba_map],
+                           native_mask=tmp_g,
+                           resolve_equal_probability=True)
+        tmp_c = context.temporary('gz compressed NIFTI-1 image')
+        context.runProcess('computeOneTissueMask',
+                           native_prob_map=self.csf_proba_map,
+                           others_prob_maps=[self.grey_proba_map, self.white_proba_map,
+                                             self.skull_proba_map, self.scalp_proba_map],
+                           native_mask=tmp_c,
+                           resolve_equal_probability=True)
+        white = aims.read(tmp_w.fullPath(), border=2)
+        grey = aims.read(tmp_g.fullPath(), border=2)
+        csf = aims.read(tmp_c.fullPath(), border=2)
```

Comment:
Maybe border not usefull?

Reply:
I need it to do the morpho math but now that I think about it I don't really know why 2, it seems to work with 1 @denisri do you have some explanation ?

Reply:
A border of 1 voxel should be enough for the morphomath algorithms which propagate 1 voxel at a time.

Reply:
Ok that what I thought but as every single example I could read put 2 it made me doubt.